dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,532 @@
+    so a rough idea of parameters to sample works.
+
+    For instance, let's say about 200 or 300 hyper-parameters need to be tested
+    to effectively search the possible hyper-parameters and estimators need a
+    little less than 100 epochs to converge. This will mean the dataset should
+    have chunks that are about 1/3rd of the entire dataset, and ``max_iter`` is
+    between 200 and 300.
+
+    If the search space is larger or more complex, choose to evaluate more
+    hyper-parameters. Increase the parameters to sample by a factor of 2 by:
+
+    * increasing ``params_to_sample`` by a factor of 2
+    * decreasing ``chunk_size`` by a factor of 2.
+
+    There are some limitations to the `current` implementation of Hyperband:
+
+    1. The full dataset is requested to be in distributed memory
+    2. The testing dataset should fit in the memory of a single worker
+
+    References
+    ----------
+    .. [1] "Hyperband: A novel bandit-based approach to hyperparameter
+           optimization", 2016 by L. Li, K. Jamieson, G. DeSalvo, A.
+           Rostamizadeh, and A. Talwalkar.  https://arxiv.org/abs/1603.06560
+
+    """
+
+    def __init__(
+        self,
+        estimator,
+        parameters,
+        max_iter=81,
+        aggressiveness=3,
+        test_size=None,
+        patience=False,
+        tol=1e-3,
+        random_state=None,
+        scoring=None,
+    ):
+        self.max_iter = max_iter
+        self.aggressiveness = aggressiveness
+
+        super(HyperbandSearchCV, self).__init__(
+            estimator,
+            parameters,
+            max_iter=max_iter,
+            patience=patience,
+            tol=tol,
+            test_size=test_size,
+            random_state=random_state,
+            scoring=scoring,
+        )
+
+    def _get_SHAs(self, brackets):
+
+        patience = _get_patience(self.patience, self.max_iter, self.aggressiveness)
+
+        # This is the first time self.random_state is used after
+        # HyperbandSearchCV.fit is called.
+        seed_start = check_random_state(self.random_state).randint(2 ** 31)
+        self._SHA_seed = seed_start
+
+        # These brackets are ordered by adaptivity; bracket=0 is least adaptive
+        SHAs = {
+            b: SuccessiveHalvingSearchCV(
+                self.estimator,
+                self.parameters,
+                n_initial_parameters=n,
+                n_initial_iter=r,
+                aggressiveness=self.aggressiveness,
+                max_iter=self.max_iter,
+                patience=patience,
+                tol=self.tol,
+                test_size=self.test_size,
+                random_state=seed_start + b if b != 0 else self.random_state,
+                scoring=self.scoring,
+            )
+            for b, (n, r) in brackets.items()
+        }
+        return SHAs
+
+    @gen.coroutine
+    def _fit(self, X, y, **fit_params):
+        X = self._check_array(X)
```

Comment:
Small refactor:

Define `BaseIncrementalSearchCV._validate_parameters(X, y) -> X, y, scorer`. Then reuse in all subclasses, including here.

Reply:
It would also check things like `self.max_iter < 1`, etc.

Reply:
Added.